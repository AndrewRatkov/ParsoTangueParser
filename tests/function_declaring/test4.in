def int fib(int x)
if x < 2 then
return x;
else
return fib(x - 1) + fib(x - 2);
fi
enddef