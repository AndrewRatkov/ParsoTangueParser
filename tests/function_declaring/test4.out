[DEF FUN fib(INTEGER x)=>INT]
|   
+-->IF
    |
    +--->[INT,< ]------->[INT,x]
    |        |
    |        +---------->[INT,2]
    |
    THEN
    |
    +--->[RETURN]--->[INT,x]
    |
    ELSE
    |
    +--->[RETURN]--->[INT,+ ]------->[CALL INT fib]------->[INT,- ]------->[INT,x]
    |                    |                                     |
    |                    |                                     +---------->[INT,1]
    |                    |
    |                    +---------->[CALL INT fib]------->[INT,- ]------->[INT,x]
    |                                                          |
    |                                                          +---------->[INT,2]
    |
    FI
